modules/uncertainty/R/generate_joint_ensemble_design.R

```diff
@@ -3,12 +3,155 @@
 #' all sites in a multi-site run. This function generates sample indices that are shared across sites to ensure consistent parameter sampling.
 #'
 ##' @param settings A PEcAn settings object containing ensemble configuration
+##' @param sobol for generating inputs for sobol
 ##' @param ensemble_size Integer specifying the number of ensemble members
 ##' @return  A list containing ensemble samples and indices
 ##' 
 ##' @export
 
-generate_joint_ensemble_design <- function(settings, ensemble_size ) {
+generate_joint_ensemble_design <- function(settings, ensemble_size, sobol = FALSE ) {
+  
+  ens.sample.method <- settings$ensemble$samplingspace$parameters$method
+  design_list <- list()
+  sampled_inputs <- list()
+  posterior.files = rep(NA, length(settings$pfts))
+  samp <- settings$ensemble$samplingspace
+  parents <- lapply(samp, '[[', 'parent')
+  order <- names(samp)[lapply(parents, function(tr) which(names(samp) %in% tr)) %>% unlist()]
+  samp.ordered <- samp[c(order, names(samp)[!(names(samp) %in% order)])]
+  
+  
+  # Sample parameters
+  PEcAn.uncertainty::get.parameter.samples(settings, ensemble.size = ensemble_size, posterior.files, ens.sample.method)
+  
+  
+  # Load samples from file
+  samples.file <- file.path(settings$outdir, "samples.Rdata")
+  samples <- new.env()
+  if (file.exists(samples.file)) {
+    load(samples.file, envir = samples)
+    if (!is.null(samples$ensemble.samples)) {
+      # Just a placeholder: extract representative trait index per ensemble member
+      # You may want to flatten or select indices per trait
+      design_list[["param"]] <- seq_len(ensemble_size)
+    } else {
+      PEcAn.logger::logger.warn("ensemble.samples not found in samples.Rdata")
+    }
+  } else {
+    PEcAn.logger::logger.error(samples.file, "not found, this file is required")
+  }
+  design_matrix<- data.frame(design_list)
+  
+  
+  
+  
+ if(sobol){
```

Comment:
My suggestion by Slack, which I will repeat here, is that the if(sobol) should come **AFTER THE design_matrix IS GENERATED**.  There's no way to write code that generates a Sobol design based on the design_matrix BEFORE the design_matrix is generated.